### PR TITLE
Add casting utils

### LINKED ISSUE #835: stop unsafe casting from u256 to felt/address/u128

### PR DESCRIPTION
**Pull request by ClementWalter**
Time spent on this PR: 2

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Some casting utils in `utils.utils.Helpers`, sometimes redundant, sometimes with non consistant naming, sometime very expensive.

Resolves #835

## What is the new behavior?

New utils in dedicated files with EVM consistent naming and signature.

### CODE REVIEW

File: src/kakarot/instructions/environmental_information.cairo [hunk 1]
```diff
@@ -167,63 +112,34 @@ namespace EnvironmentalInformation {
...
 
         let (sliced_calldata: felt*) = alloc();
-
-        let calldata: felt* = ctx.call_context.calldata;
-        let calldata_len: felt = ctx.call_context.calldata_len;
-
-        // read calldata at offset
-        let sliced_calldata: felt* = Helpers.slice_data(
-            data_len=calldata_len, data=calldata, data_offset=calldata_offset.low, slice_len=32
+        slice(
```
Comment (Eikix) [inline src/kakarot/instructions/environmental_information.cairo:118]: I wonder why we have to make a copy of the slice of the calldata inside sliced_calldata first,

We could technically accumulate a uint256 by reading the ctx.call_context.calldata starting at offset.low, for 32 indexes?

Though i understand it's not the purpose of this PR

**Review by Eikix — DISMISSED** [review]
lgtm

File: src/kakarot/instructions/environmental_information.cairo (hunk 1 above)
Comment (ClementWalter) [reply src/kakarot/instructions/environmental_information.cairo:118]: I had a big PRs with a lot of changes and was not able to easily understand why I had newly failing test cases, so I've closed it and created a step by step PR, indeed not challenging to much the existing but mainly integrating the new utils. For the record, I've now solved some "maximum offset value exceeded", see https://github.com/kkrt-labs/ef-tests/pull/610.

Then I'll move on to the refacto of the logic

**Review by Eikix — DISMISSED** [review]
lgtm

**Pull request merged**